docs/apple-silicon.md

```diff
@@ -0,0 +1,176 @@
+## Notes: CIEL on Apple Silicon
+
+We need to adapt requirements to our environment, and since Apple Silicon is
+different from other Unix-like OSes, also in terms of where libraries
+and their header files are located, i.e. installed via `homebrew`, we
+need to perform a few additional steps to have CIEL working as it should.
+
+First and foremost, make sure that you are running up to date versions of
+Common Lisp implementation (only tested with SBCL), and `asdf`. Also
+make sure to have added Ultralisp to Quicklisp dists.
+
+### SBCL (homebrew)
+
+```example
+$ brew install sbcl
+$ curl -o /tmp/quicklisp.lisp https://beta.quicklisp.org/quicklisp.lisp
+$ sbcl --load /tmp/quicklisp.lisp
+```
+
+```commonlisp
+(quicklisp-quickstart:install)
+(sb-ext:exit)
+```
+
+```example
+$ rm /tmp/quicklisp.lisp
+```
+
+### SBCL via Roswell (homebrew)
+
+```example
+$ brew update && brew install roswell
+$ ros # installs sbcl by default
+$ ros run # verify by starting REPL
+```
+
+### ASDF
+```example
+$ mkdir -p ~/common-lisp && cd ~/common-lisp
+$ git clone https://gitlab.common-lisp.net/asdf/asdf.git
```

Comment:
We need to install the latest asdf to sbcl. Because the sbcl which installed through homebrew has the asdf version 3.3.1 (I don't know why sbcl don't update it for years). So the `make ql-deps` will return the error that the asdf version is too old (it needs at least 3.3.4). 

I reinstall asdf to sbcl every time the sbcl updated on homebrew. 

```
cd ~/common-lisp/asdf
make
./tools/asdf-tools install-asdf sbcl
```

then the `sbcl --script check-asdf-version.lisp` in ciel folder will return the newest version (3.3.7.1)

Reply:
Hmm... I did not encounter that problem. The only thing I did with the homebrew `sbcl` install is to make sure quicklisp is installed and initialised via `.sbclrc`, then clone `asdf` into `~/common-lisp`, or whatever other location one has chosen to use. I must be missing something, because I end up with `(asdf:asdf-version) ; 3.3.7.1`. Am I going about it the wrong way?

Reply:
I reinstalled SBCL through Homebrew, and the ASDF version was reverted to `3.3.1`. I had studied the issue before, I believe the problem lies in this [code](https://github.com/Homebrew/homebrew-core/blob/9de737780e3d0003aa2270a84a243b70e236ad20/Formula/s/sbcl.rb#L57-L60), which includes the `contrib/` folder (specifically located at `/opt/homebrew/Cellar/sbcl/2.4.8/lib/sbcl/sbclrc`). The [SBCL version of ASDF](https://github.com/sbcl/sbcl/tree/8ad09201276ed3ba8138551efcd92dcf1bb7920f/contrib/asdf) has been stuck at version 3.3.1 for years.

I maybe miss something. I would love to sbcl use newest asdf automatically rather than update every time by myself.

Reply:
Yea.. I think I read this issue in sbcl maillist too. I am thinking about give some contributions to homebrew. But I know nothing about ruby 🙃.